src/hdf5_time_series_storage.jl

```diff
@@ -667,114 +672,58 @@ end
-        data_read = retransform_hdf_array(dataset[rows, columns, :, :], type)
-        for (i, it) in
-            enumerate(range(start_time; length = length(columns), step = interval))
-            data[it] = @view data_read[1:length(rows), i]
-        end
-    end
+    data = retransform_hdf_array(dataset[rows], type)
     return data
 end
 
 function get_hdf_array(
     dataset,
-    type::Union{Type{<:CONSTANT}, Type{LinearFunctionData}},
+    T,
     rows::UnitRange{Int},
 )
-    data = retransform_hdf_array(dataset[rows], type)
-    return data
-end
-
-function get_hdf_array(dataset, type::Type{QuadraticFunctionData}, rows::UnitRange{Int})
-    data = retransform_hdf_array(dataset[rows, :, :], type)
-    return data
-end
-
-function get_hdf_array(dataset, type::Type{PiecewiseLinearPointData}, rows::UnitRange{Int})
-    data = retransform_hdf_array(dataset[rows, :, :, :], type)
+    colons = repeat([:], ndims(dataset) - 1)
+    data = retransform_hdf_array(dataset[rows, colons...], T)
     return data
 end
 
 function retransform_hdf_array(data::Array, ::Type{<:CONSTANT})
     return data
 end
 
-function retransform_hdf_array(data::Array, ::Type{LinearFunctionData})
-    return LinearFunctionData.(data)
-end
-
-function retransform_hdf_array(data::Array, T::Type{QuadraticFunctionData})
-    row, column, tuple_length = get_data_dims(data, T)
-    if isnothing(column)
-        t_data = Array{Tuple{Float64, Float64}}(undef, row)
-        for r in 1:row
-            t_data[r] = tuple(data[r, 1:tuple_length]...)
-        end
-    else
-        t_data = Array{Tuple{Float64, Float64}}(undef, row, column)
-        for r in 1:row, c in 1:column
-            t_data[r, c] = tuple(data[r, c, 1:tuple_length]...)
-        end
-    end
-    return _quadratic_from_tuple.(t_data)
-end
-
-function retransform_hdf_array(data::Array, T::Type{PiecewiseLinearPointData})
-    row, column, tuple_length, array_length = get_data_dims(data, T)
-    if isnothing(column)
-        t_data = Array{Vector{Tuple{Float64, Float64}}}(undef, row)
-        for r in 1:row
-            tuple_array = Array{Tuple{Float64, Float64}}(undef, array_length)
-            for l in 1:array_length
-                tuple_array[l] = tuple(data[r, 1:tuple_length, l]...)
-            end
-            t_data[r] = tuple_array
-        end
-    else
-        t_data = Array{Vector{Tuple{Float64, Float64}}}(undef, row, column)
-        for r in 1:row, c in 1:column
-            tuple_array = Array{Tuple{Float64, Float64}}(undef, array_length)
-            for l in 1:array_length
-                tuple_array[l] = tuple(data[r, c, 1:tuple_length, l]...)
-            end
-            t_data[r, c] = tuple_array
-        end
-    end
-    return PiecewiseLinearPointData.(t_data)
+function retransform_hdf_array(
+    data::Array,
+    T::Union{Type{LinearFunctionData}, Type{QuadraticFunctionData}},
+)
+    length_req = fieldcount(get_raw_data_type(T))
+    (size(data)[end] == length_req) || throw(
```

Comment:
I don't find anything objectionable about this pattern. However, I do want consistency in our error checks. So, either we always do it this way or always do it the other.
```
size(data)[end] != length_req && throw
```
or 
```
if size(data)[end] != length_req
    throw ...
end
```

Reply:
Is it the short-circuiting operator rather than the `if` block you take issue with or the fact that I'm using `x || throw` rather than `!x && throw`? Happy to change these to the existing convention.

Reply:
I'm fine with leaving this as-is.

Reply:
Removed all the occurrences of the `||` pattern